velox/exec/prefixsort/PrefixSortEncoder.h

```diff
@@ -228,4 +237,126 @@ FOLLY_ALWAYS_INLINE void PrefixSortEncoder::encodeNoNulls(
   encodeNoNulls(value.getNanos(), dest + 8);
 }
 
+class PrefixSortLongDecimalToIntEncoder : public PrefixSortEncoder {
+ public:
+  PrefixSortLongDecimalToIntEncoder(
+      bool ascending,
+      bool nullsFirst,
+      int precision,
+      int scale)
+      : PrefixSortEncoder(ascending, nullsFirst),
+        precision_(precision),
+        scale_(scale),
+        newScale_(newPrecision_ - (precision_ - scale_)) {
+    VELOX_DCHECK(precision - scale <= ShortDecimalType::kMaxPrecision);
+  };
+
+  FOLLY_ALWAYS_INLINE void encode(std::optional<int128_t> value, char* dest)
+      const override {
+    if (value.has_value()) {
+      dest[0] = nullsFirst_ ? 1 : 0;
+      int64_t newValue;
+      auto status = DecimalUtil::rescaleWithRoundUp<int128_t, int64_t>(
+          value.value(),
+          precision_,
+          scale_,
+          newPrecision_,
+          newScale_,
+          newValue);
+      if (FOLLY_UNLIKELY(!status.ok())) {
```

Comment:
Why here we can consider status not ok as an expected behavior ？
Can you add a comment here to explain?

Reply:
Spark implementation https://github.com/apache/spark/blob/branch-3.5/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/SortOrder.scala#L172

Background
https://issues.apache.org/jira/browse/SPARK-40089

Reply:
It would be nice to add an ut for these case, or do I miss it ?

Reply:
@mbasmanova I think the diff with velox  is here: the case when !status.ok() may generate different result.